src/App.tsx

```diff
@@ -1,24 +1,391 @@
+
+    todos.forEach(todo => {
+      if ((allTodosCompleted && todo.completed)
+        || (!allTodosCompleted && !todo.completed)) {
+        checkTodo(todo, !todo.completed);
+      }
+    });
+  };
+
+  const handleDoubleClick = (selectedTodo: Todo) => {
+    setIsTodoEditing(true);
+    setTodoEditing(selectedTodo);
+    setNewTitle(selectedTodo.title);
+  };
+
+  const handleTodoUpdated = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setNewTitle(e.target.value);
+  };
+
+  const changeTodoTitle = (event: React.FormEvent, editedTitle: string) => {
+    event.preventDefault();
+
+    if (!todoEditing) {
+      return;
+    }
+
+    if (!editedTitle) {
+      deleteTodo(todoEditing.id);
+    }
+
+    if (editedTitle !== todoEditing.title) {
+      updateTodo({
+        ...todoEditing,
+        title: editedTitle,
+      });
+    }
+
+    setIsTodoEditing(false);
+  };
+
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">React Todo App - Add and Delete</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <header className="todoapp__header">
+          <button
+            type="button"
+            className={classNames('todoapp__toggle-all', {
+              active: activeTodos.length === 0,
+            })}
+            onClick={handleToggleAll}
+          />
+
+          <form onSubmit={(event) => addTodo(event, query)}>
+            <input
+              type="text"
+              className={classNames(
+                'todoapp__new-todo',
+                { 'todoapp__new-todo__load': isTodoLoaded },
+              )}
+              placeholder="What needs to be done?"
+              onChange={handleTodoInput}
+              value={query}
+            />
+          </form>
+        </header>
+
+        {todos.length > 0 && (
+          <section className="todoapp__main">
+            {visibleTodos.map(todo => {
+              const { id, title, completed } = todo;
+
+              return (
+                <div
+                  className={classNames('todo', { completed })}
+                  key={id}
+                >
+                  <label className="todo__status-label">
+                    <input
+                      type="checkbox"
+                      className="todo__status"
+                      checked={completed}
+                      onClick={() => checkTodo(todo)}
+                    />
+                  </label>
+
+                  {(isTodoEditing && todoEditing?.id === id) ? (
+                    <form onSubmit={
+                      (event) => changeTodoTitle(event, newTitle)
```

Comment:
same for anonymous arrow function in jsx
make handler and pass referense here
check all places with anonymous arrow functions in jsx